src/System/extract_state.rs

```diff
@@ -73,25 +73,98 @@ fn build_location_tuple(
     machine: &SystemRecipe,
     system: &TransitionSystemPtr,
 ) -> Result<LocationTuple, String> {
-    let location_id = get_location_id(&mut locations.iter(), machine);
+    let out = is_universal_or_inconsistent_input(locations, machine);
+    let system_locations = system.get_all_locations();
 
-    system
-        .get_all_locations()
+    //If the location is universal
+    if out.0 {
+        loc_find_and_then(
+            system_locations,
+            &|loc: &&LocationTuple| matches!(loc.loc_type, LocationType::Universal),
+            &|loc: &LocationTuple| Some(loc.to_owned()),
+            "Could not find universal location in the transition system",
+        )
+    // If the location is inconsistent
+    } else if out.1 {
+        loc_find_and_then(
+            system_locations,
+            &|loc| matches!(loc.loc_type, LocationType::Inconsistent),
+            &|loc| Some(loc.to_owned()),
+            "Could not find inconsistent location in the transition system",
+        )
+    // If the location is normal
+    } else {
+        let location_id = get_location_id(&mut locations.iter(), machine);
+
+        loc_find_and_then(
+            system_locations,
+            &|loc| loc.id.compare_partial_locations(&location_id),
+            &|location_tuple| {
+                Some(if !location_id.is_partial_location() {
+                    location_tuple.clone()
+                } else {
+                    LocationTuple::create_partial_location(location_id.clone())
+                })
+            },
+            format!("{} is not a location in the transition system", location_id).as_str(),
+        )
+    }
+}
+
+/// Checks if the input [LocationTuple] is of type `LocationType::Universal` or `LocationType::Inconsistent`.
+/// Returns (`bool 0: is_universal`, `bool 1: is_inconsistent`)
+fn is_universal_or_inconsistent_input(locations: &[&str], machine: &SystemRecipe) -> (bool, bool) {
```

Comment:
Is it possible to give the returned tuple names for the fields?
so that instead of out.0 and out.1, it gives out.is_universal and out.is_inconsistent

Reply:
apparantly not possible in rust :(